roles/jd-server/src/lib/job_declarator/message_handler.rs

```diff
@@ -61,6 +62,61 @@ impl ParseClientJobDeclarationMessages for JobDeclaratorDownstream {
     }
 
     fn handle_declare_mining_job(&mut self, message: DeclareMiningJob) -> Result<SendTo, Error> {
+        {
+            // Clone the old declared mining job to retain its data
+            let old_declare_mining_job_ = self.declared_mining_job.0.clone();
+
+            // If there is an old declared mining job, remove its transactions from the mempool
+            if let Some(old_job) = &old_declare_mining_job_ {
+                // Retrieve necessary data from the old job
+                let old_tx_hash_list = old_job.tx_short_hash_list.inner_as_ref();
+                let old_nonce = old_job.tx_short_hash_nonce;
+
+                // Get the mempool of the old transaction IDs
+                let old_short_id_mempool =
+                    match self.mempool.safe_lock(|x| x.to_short_ids(old_nonce)) {
+                        // If successfully retrieved, store it
+                        Ok(Some(mempool)) => mempool,
+                        // If the mempool is not found, return an error
+                        Ok(None) => {
+                            return Err(roles_logic_sv2::Error::PoisonLock(
```

Comment:
Do not return a poison lock here

Reply:
What error class should i return please? A regular error string?

Reply:
I thought about.
[Few lines](https://github.com/stratum-mining/stratum/blob/main/roles/jd-server/src/lib/job_declarator/message_handler.rs#L82) below there already a double unwrap. I think that fix that is out of the scope of this PR, so imo you can double unwrap, as long as you open an issue for that.
So
```
                // Get the mempool of the old transaction IDs
                let old_short_id_mempool = self
                    .mempool
                    .safe_lock(|x| x.to_short_ids(old_nonce))
                    .unwrap()
                    .unwrap();
```